hoot-core/src/main/cpp/hoot/core/elements/MapProjector.cpp

```diff
@@ -563,28 +539,27 @@ void MapProjector::project(const std::shared_ptr<OsmMap>& map,
 }
 
 void MapProjector::project(const std::shared_ptr<Geometry>& g,
-  const std::shared_ptr<OGRSpatialReference>& srs1,
-  const std::shared_ptr<OGRSpatialReference>& srs2)
+                           const OGRSpatialReferencePtr& srs1,
+                           const OGRSpatialReferencePtr& srs2)
 {
   OGRCoordinateTransformation* t(OGRCreateCoordinateTransformation(srs1.get(), srs2.get()));
 
   if (t == nullptr)
-  {
     throw HootException(QString("Error creating transformation object: ") + CPLGetLastErrorMsg());
-  }
 
   LOG_TRACE("Reprojecting map from: " << toWkt(srs1) << " to " << toWkt(srs1) << "...");
 
   ReprojectCoordinateFilter filter(t);
   g->apply_rw(&filter);
+  //  NOTE: It is very important that the geometry is marked as changed after if it reprojected
+  g->geometryChanged();
```

Comment:
This is it!  This is the piece that was missing.  When `Geometry::getEnvelope()` is called, the envelope is stored.  The `ReprojectCoordinateFilter` class reprojects the nodes but doesn't update the buffered envelope.  Calling `Geometry::geometryChanged()` invalidates the envelope and a new one is created by the next call to `Geometry::getEnvelope()`.